### DISCUSSION TITLE
Comparing TCP and UDP, technical report.

### DISCUSSION THREAD

**Discussion by Nordin-S** [dannagle/PacketSender#241] [General]
Hi,
I'm a student doing a networking course during a Bachelor of Science program.
We have a technical repport to do where we are ment to compare both theoretically and technically two applications, devices or technologies. I have decided to compare UDP and TCP, researching how to compare and test the technical part I've stumbled upon this tool which looks promising. I was wondering if my assumption is right that this tool could serve my needs and if so do you have any suggestion in regards to what technical aspects could be compared with this tool. If not I would be grateful to be pointed into the right direction. 

Thanks in advance!
Kind regards

**Comment by dannagle** [comment]
Hello, I think Packet Sender could be very helpful in your studies. Packet Sender is actually already used in some college classrooms and as part of some senior design projects. 

Some notes, in no particular order:
- The default installation comes bundled with a bunch of sample packets targeting internet servers. Click Send for a quick test. These samples cover various internet protocols, such as DNS and NTP through UDP. It also has examples such as HTTP, Telnet, and FTP through TCP.
- SSL is encryption on top of TCP. There are example packets for this for SMTP, HTTPS
- Packet Sender also comes with UDP, TCP, and SSL servers. Port numbers are in the bottom right. You can run tests by having 2 talk to each other.
- Something unique with TCP is the idea of persistent connections. Packet Sender has a checkbox you can test this. College labs sometimes use this to make an ad-hoc chat app for the students.
- Broadcast is unique feature for UDP. You can test this with multiple Packet Senders. An extremely insightful test would be to go to Settings and have a Packet Sender auto-reply. Congrats. You just built a discovery protocol.
- Multicast is unique to UDP. It is a bit more difficult to set up, but Packet Sender does support multicast join. 

I hope that is enough to get you started. Feel free to post questions.

**Reply by Nordin-S** [reply]
Thank you for the response!
I have played around and gotten hang of it a bit, it's pretty awesome I must say.

Have unfortunately not been able to make two packet senders talk to one another. Got the iOS app version and have both tried with Wi-Fi and cellular to get a hold of it's ip and also using both port 5005 and other ports to send packets from my laptop to it, no packets show up.

Any ideas on how to get that to work? I think that feature is most relevant to me. Will se how this feature works in a few days between the laptop and desktop PC.

**Reply by dannagle** [reply]
I doubt you will be able to send UDP from your WiFi desktop to your phone over cellular. Routers like to filter UDP. Your best hope is to have both connected to the same WiFi router. 

You may be hitting firewall issues. I would experiment with 2 desktop versions before attempting any mobile tests. There are far more troubleshooting tools available for desktop. 

As usual, the first debug steps are to make sure the sample packets work. That's why they are there.
